### PR TITLE
Add type hints to sybil.integration.unittest

### PR DESCRIPTION
**Pull request by adamtheturtle**
This resolves all type errors when running `mypy --strict sybil/integration/unittest.py`.

### CODE REVIEW

File: sybil/integration/unittest.py [hunk 1]
```diff
@@ -1,22 +1,28 @@
-from typing import TYPE_CHECKING
+from __future__ import annotations
```
Comment (cjw296) [inline sybil/integration/unittest.py:1]: I don't like this, please don't use it.

File: sybil/integration/unittest.py [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
 if TYPE_CHECKING:
+    from unittest.loader import TestLoader
+
     from ..sybil import Sybil
+    from sybil.example import Example
 
 
 class TestCase(BaseTestCase):
 
-    sybil = namespace = None
```
[the remark this answers is not in the document]
Comment (cjw296) [reply sybil/integration/unittest.py:10]: Yeah, if the tests pass we're all good.

**Comment by adamtheturtle** [comment]
Thank you for the review @cjw296 . I have moved imports out of `if TYPE_CHECKING`, and I have quoted some annotations so that I could remove `from __future__ import annotations`. Please take another look

File: sybil/integration/unittest.py [hunk 3]
```diff
@@ -1,22 +1,28 @@
-from typing import TYPE_CHECKING
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable, Optional
```
[the remark this answers is not in the document]
Comment (adamtheturtle) [reply sybil/integration/unittest.py]: Ok I have changed this to add nothing into `if TYPE_CHECKING:	`.

File: sybil/integration/unittest.py (hunk 1 above)
Comment (adamtheturtle) [reply sybil/integration/unittest.py]: I have changed this to use quoted annotations rather than `from __future__ import annotations`.

**Pull request merged**